qiskit/transpiler/passmanager.py

```diff
@@ -133,9 +165,23 @@ def run(self, circuit):
         del circuit
         self.reset()  # Reset passmanager instance before starting
 
+        count = 0
         for passset in self.working_list:
             for pass_ in passset:
+                if self.callback:
+                    start_time = time()
                 dag = self._do_pass(pass_, dag, passset.options)
+                if self.callback:
+                    end_time = time()
+                if self.callback:
+                    run_time = end_time - start_time
+                    # Deepcopy dag to ensure reference doesn't get updated
+                    # if callback func uses the dag after the loop iteration
+                    # continues
+                    self.callback(pass_=pass_, dag=copy.deepcopy(dag),
```

Comment:
I would say one of the advantages of the callback is that people can "inject code", meaning get a handle to an intermediate dag, modify it, and continue. A kind of interactive debugging. So I don't think we should prevent that here. It's the user's responsibility what to do with the dag.

Reply:
Sure, I did the deepcopy mostly for the tests, but I agree it's probably better to just do the deepcopy in the test's callback function instead of here.